crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -532,12 +532,12 @@ impl RpcServerConfig {
             .http_addr
             .unwrap_or(SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::UNSPECIFIED, DEFAULT_HTTP_RPC_PORT)));
 
-        let mut local_addr = None;
+        let mut http_local_addr = None;
 
         let http_server = if let Some(builder) = self.http_server_config {
             let server = builder.build(http_socket_addr).await?;
-            if local_addr.is_none() {
-                local_addr = server.local_addr().ok();
+            if http_local_addr.is_none() {
```

Comment:
this check is now redundant since we have dedicated addresses

we should use `let http_local_addr = server.local_addr().ok()`